bookmarks/api/serializers.py

```diff
@@ -111,6 +112,17 @@ def create(self, validated_data):
         # title and description to be populated automatically when left empty
         if not disable_scraping:
             bookmarks.enhance_with_website_metadata(saved_bookmark)
+
+        # override default timestamps, if specified
+        added = 'date_added' in validated_data
+        modified = 'date_modified' in validated_data
+        if added or modified:
+            if added:
+                saved_bookmark.date_added = validated_data['date_added']
+            if modified:
+                saved_bookmark.date_modified = validated_data['date_modified']
+            saved_bookmark.save()
```

Comment:
Instead of saving the bookmark twice it would probably make more sense to change `create_bookmark` so that it only initializes the dates if the new bookmark doesn't already specify them.

Adding some test cases would also be helpful.

Reply:
yes, fair enough.  Will do!